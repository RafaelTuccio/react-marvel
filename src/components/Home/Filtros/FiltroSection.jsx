import ic_heroi from "../../assets/ic_heroi.svg"
import toggleOff from "../../assets/toggle_off.svg"
import toggleOn from "../../assets/toggle_on.svg"
import favoritoOff from "../../assets/favorito_02.svg"
import favoritoOn from "../../assets/favorito_01.svg"


const Filter = (props) => {
    let { favorito, filtro, getFiltro, showFavorito, count, total} = props;

    return(
       
            <div className="filtros-paginacao"> 
                <div className="paginacao">
                    <span>Encontrados {count} de {total} her√≥is</span>
                </div>
                <div className="filtros">
                    <img src={ic_heroi} alt="" />
                    <span className="filtro-span">Ordernar por nome - A/Z</span>
                    <div onClick={()=> {
                        getFiltro()
                    }}>
                        {
                            filtro === false ?
                            <img src={toggleOff} alt="botao desligado" />
                            : <img src={toggleOn} alt="botao ligado" />
                        }
                    </div>
                </div>
                <div className="favoritos">
                    <div onClick={() => {
                        showFavorito()
                    }}>
                        {
                            favorito === false ?
                            <img src={favoritoOff} alt="coracao branco" />
                            : <img src={favoritoOn} alt="coracao vermelho" />
                        }
                    </div>
                        <span className="favoritos-span">Somente favoritos</span>
                </div>
                
            </div>
    );
}

export default Filter;