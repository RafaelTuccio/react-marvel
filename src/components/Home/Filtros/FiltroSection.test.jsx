import { render, screen } from '@testing-library/react';
import FiltroSection from './FiltroSection';

describe("<FiltroSection />", () => {
    it("deve renderizar na tela normalmente", () => {
        const { debug } = render(<FiltroSection />)

        debug();
    })

    it("deve dar match com o snapshot", () => {
        
        const { container } = render(<FiltroSection />);
        expect(container.firstChild).toMatchSnapshot();
    })


})