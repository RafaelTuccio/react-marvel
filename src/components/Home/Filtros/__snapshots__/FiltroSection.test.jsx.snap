// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FiltroSection /> deve dar match com o snapshot 1`] = `
<div
  class="filtros-paginacao"
>
  <div
    class="paginacao"
  >
    <span>
      Encontrados 20 her√≥is
    </span>
  </div>
  <div
    class="filtros"
  >
    <img
      alt=""
      src="ic_heroi.svg"
    />
    <span
      class="filtro-span"
    >
      Ordernar por nome - A/Z
    </span>
    <div>
      <img
        alt="botao ligado"
        src="toggle_on.svg"
      />
    </div>
  </div>
  <div
    class="favoritos"
  >
    <div>
      <img
        alt="coracao vermelho"
        src="favorito_01.svg"
      />
    </div>
    <span
      class="favoritos-span"
    >
      Somente favoritos
    </span>
  </div>
</div>
`;
